iroh-net/src/relay/iroh_relay.rs

```diff
@@ -800,21 +793,27 @@ mod tests {
         assert!(body.is_empty());
     }
 
+    #[tokio::test]
+    async fn test_relay_client_legacy_route() {
+        let _guard = iroh_test::logging::setup();
+        let server = spawn_local_relay().await.unwrap();
+        let relay_url = format!("http://{}", server.http_addr().unwrap());
+        let relay_url: RelayUrl = relay_url.parse().unwrap();
+
+        // set up client a
+        let secret_key = SecretKey::generate();
+        let resolver = crate::dns::default_resolver().clone();
+        let (client, _) = ClientBuilder::new(relay_url)
+            .relay_path("/derp") // Try the legacy relay path for backwards compatibility
+            .build(secret_key, resolver);
+
+        client.ping().await.unwrap();
+    }
```

Comment:
Setup this fairly simple smoke test. Does that cover what you were thinking? @dignifiedquire 
It's not doing the whole "two clients connected" thing, but it does exercise the relay to make sure it actually expects to receive `Frame`s and responds.

Reply:
You could argue that you only care that the `/derp` endpoint responds with the right HTTP response for the test.  That would allow you to write a test with a few request calls instead of checking the entire protocol works on `/derp`.  The reasoning is that we already test the implementation on the `/relay` endpoint and the handler is the same anyway.  It would mean you don't need the `.relay_path` on the builder which is a bit unfortunate I think, as it would encourage people to change in on their setups.

Reply:
Because once you start providing that option, you open the door to requests for the other paths like `generate_204` etc.  Which just seems messy for very little value.  (maybe someone comes up with a really good usecase and I'll change my mind.  but let's wait for that)

Reply:
Agreed. I want to avoid the `relay_path` as much as possible. I was worried I'm not testing the `/derp` endpoint enough.

I'll change the PR to remove the `relay_path` change & only test the HTTP upgrade on `/derp`.